src/TodoList.coffee

```diff
@@ -0,0 +1,53 @@
+define 'TodoList',
+[
+    './ViewInterface.coffee'
+],
+(ViewInterface) ->
+    class TodoList extends ViewInterface
+        items = ['First todo']
+        todoListSelector = document.querySelector('.todo-list')
+        init: ->
+            _render()
+            document.querySelector('.todo-list-form').addEventListener('submit', _addItem.bind(@))
+            document.addEventListener('click', _deleteItem.bind(@))
+            document.addEventListener('click', _editItem.bind(@))
+            todoListSelector.addEventListener('DOMNodeRemoved', _onDestroy.bind(@))
+        _onDestroy = ->
+            document.querySelector('.todo-list-form').removeEventListener('submit', _addItem.bind(@))
+            document.removeEventListener('click', _deleteItem.bind(@))
+            document.removeEventListener('click', _editItem.bind(@))
+        _editItem = (e) ->
+            { target } = e
+            if (target.classList.contains('todo-item-text'))
+                text = target.parentElement.dataset.key
+                target.innerHTML = "<input name='edit-text' value='#{text}'/>"
+                inputSelector = target.querySelector('input')
+                inputSelector.focus()
+                inputSelector.onblur = (e) ->
+                    items[items.indexOf(text)] = e.target.value
+                    _render()
+        _deleteItem = (e) ->
+            { target } = e
+            if (target.classList.contains("delete-btn"))
```

Comment:
move target.classList.contains("delete-btn") to variable isDeleteButton.
You actually can use 

```
document.querySelector('.delete-btn').addEventListener
````

Instead of  adding click to whole document

Reply:
?